app/models/user.rb

```diff
@@ -63,6 +67,27 @@ def is_application?
     false
   end
 
+  # State helpers.
+  #
+  # A User model begins life in the "temp" state, and can then be claimed by another user
+  # who originated from an OAuth login. Upon it being claimed it will be removed and it's
+  # data merged with the claimant.
+  #
+  # A User can also be created by a one of the consumer applications as a stand-in
+  # for a person who has not yet (or may never) created an account.  In this case
+  # the User model will be in the "unclaimed" state.  When the User does signup, they
```

Comment:
It's actually in the finish user creation routine: https://github.com/openstax/accounts/blob/temp-users/app/routines/finish_user_creation.rb#L8

All that routine does is check if the user is already activated at the top and bail if it is.  The remainder of it sets the user to "activated" and will be called regardless of whether their "temp" or "unclaimed"